README-dev.md

```diff
@@ -6,36 +6,61 @@ This README is for SDK developers. If you are a plugin developer please refer to
+the API plugin developers write against.
+
+All dependencies of a plugin must be packaged with the plugin including the protobuf messages (`dvp-api`) and the wrappers.
+This is done automatically by `dvp build`.
 
-1. Make changes to SDK and appgate code. Test these changes manually. Iterate on this until you have everything working.
-2. Publish a development build of the SDK to artifactory.
-3. Update the version of the SDK specified in the app gate.
-4. Publish a review for SDK code, and also publish a "provisional" review of appgate code. Address any feedback.
-5. Push the SDK code and publish new SDK builds to our internal servers.
-6. Finalize your appgate review.
-7. Push the appgate changes
+This is what causes the slightly different workflows in development. Changes to `tools` are completely isolated from the
+Delphix Engine and wrappers changes only impact the plugin build.
 
-Not every type of change requires every step.
+## Development process
+
+At a very high level, our development process usually looks like this:
+
+1. Create a fork of the delphix/virtualization-sdk repository.
+2. Clone the forked repository.
+3. Make changes to SDK code. Test these changes manually and with unit tests. Iterate on this until you have everything working.
+4. Commit your changes and build all Python package distributions. Make sure the version number of the packages is updated appropriately.
+5. Publish Python distributions to artifactory. 
+7. Run blackbox against the newly uploaded SDK version.
+7. Publish a pull request to the delphix/virtualization-sdk once your code is ready for review.
+8. Once the pull request is approved, it will merged into delphix/virtualization-sdk repository.
 
 These steps are described in more detail below.
 
-## Background
+## Versioning
 
-There are two parts of the SDK that are important to think about separately since they have slightly different workflows.
+The SDK is shipped as five Python packages that are currently versioned and shipped together: dvp, dvp-common, dvp-libs,
+dvp-platform, and dvp-tools. 
 
-1. The `tools` package is the SDK's CLI. This aids in plugin development, testing, and distribution.
-2. `common`, `libs`, and `platform` contain what are collectively called the "wrappers". These are vanilla Python classes that abstract the Virtualization API protobuf messages (published by app-gate) away from plugin developers. These expose the API plugin developers write against.
+The first thing to do is to change the version number of all the packages. Our versioning scheme follows the rules of
+semantic versioning in order to help developers manage their "dependency hell". We use bump2version 
+(https://github.com/c4urself/bump2version) to make the version management of all five packages easier. If you don't have
+bump2version installed, simply run `pip install bump2version`. Semantic versioning rules are the following:
```

Comment:
bump2version should be specified in the requirements.txt file I think and people should install that first during development.

Reply:
I see that installing from requirements.txt is specified below, maybe this section should go after that section?

Reply:
I also think all the requirements.txt files need to have `pytest` in them

Reply:
I moved the `Versioning` section between `Wrappers changes` and `Testing`. Setting up development environment is described in the README in `tools` and referenced from `Local SDK Development` section here.

Reply:
Added `bump2version` to all requirements.txt files. `pytest` is already there.